x/checkpoint/side_handler_test.go

```diff
@@ -0,0 +1,461 @@
+	require.NoError(t, err)
+	borChainId := "1234"
+	suite.Run("Success", func() {
+		suite.contractCaller = mocks.IContractCaller{}
+
+		proposer, err := sdk.AccAddressFromHex(header.Proposer)
+		require.NoError(t, err)
+
+		// create checkpoint msg
+		msgCheckpoint := types.NewMsgCheckpointBlock(
+			proposer,
+			header.StartBlock,
+			header.EndBlock,
+			hmCommonTypes.HexToHeimdallHash(header.RootHash),
+			hmCommonTypes.HexToHeimdallHash(header.RootHash),
+			borChainId,
+		)
+
+		suite.contractCaller.On("CheckIfBlocksExist", header.EndBlock).Return(true)
+		suite.contractCaller.On("GetRootHash", header.StartBlock, header.EndBlock, uint64(1024)).Return(hmCommonTypes.HexToHeimdallHash(header.RootHash).Bytes(), nil)
+
+		result := suite.sideHandler(ctx, &msgCheckpoint)
+		// todo:   uint32(0) needs convert sdk.CodeOK
+		require.Equal(t, uint32(0), result.Code, "Side tx handler should be success")
+		require.Equal(t, abci.SideTxResultType_YES, result.Result, "Result should be `yes`")
+
+		bufferedHeader, _ := keeper.GetCheckpointFromBuffer(ctx)
+		require.Nil(t, bufferedHeader, "Should not store state")
+	})
+
+	suite.Run("No root hash", func() {
+		suite.contractCaller = mocks.IContractCaller{}
+		proposer, err := sdk.AccAddressFromHex(header.Proposer)
+		require.NoError(t, err)
+		// create checkpoint msg
+		msgCheckpoint := types.NewMsgCheckpointBlock(
+			proposer,
+			header.StartBlock,
+			header.EndBlock,
+			hmCommonTypes.HexToHeimdallHash(header.RootHash),
+			hmCommonTypes.HexToHeimdallHash(header.RootHash),
+			borChainId,
+		)
+
+		suite.contractCaller.On("CheckIfBlocksExist", header.EndBlock).Return(true)
+		suite.contractCaller.On("GetRootHash", header.StartBlock, header.EndBlock, uint64(1024)).Return(nil, nil)
+
+		result := suite.sideHandler(ctx, &msgCheckpoint)
+		// 0 needs convert sdk.CodeOK
+		require.Equal(t, uint32(0), result.Code, "Side tx handler should succeed")
+		require.Equal(t, abci.SideTxResultType_SKIP, result.Result, "Result should be `skip`")
+
+		bufferedHeader, err := keeper.GetCheckpointFromBuffer(ctx)
+		require.Error(t, err)
+		require.Nil(t, bufferedHeader, "Should not store state")
+	})
+	//
+
+	suite.Run("invalid checkpoint", func() {
+		suite.contractCaller = mocks.IContractCaller{}
+		proposer, err := sdk.AccAddressFromHex(header.Proposer)
+		require.NoError(t, err)
+		// create checkpoint msg
+		msgCheckpoint := types.NewMsgCheckpointBlock(
+			proposer,
+			header.StartBlock,
+			header.EndBlock,
+			hmCommonTypes.HexToHeimdallHash(header.RootHash),
+			hmCommonTypes.HexToHeimdallHash(header.RootHash),
+			borChainId,
+		)
+
+		suite.contractCaller.On("CheckIfBlocksExist", header.EndBlock).Return(true)
+		suite.contractCaller.On("GetRootHash", header.StartBlock, header.EndBlock, uint64(1024)).Return([]byte{1}, nil)
+
+		result := suite.sideHandler(ctx, &msgCheckpoint)
+		require.NotNil(t, result)
+		require.Equal(t, uint32(0), result.Code, "Side tx handler should not succeed")
```

Comment:
We used to use CodeOK https://github.com/maticnetwork/cosmos-sdk/blob/sidechannel/types/errors.go#L29 earlier but now this is not there, hence we are directly equating with 0. What would be the right way to do it? @jdkanani @anilCSE